### PR TITLE
Extend-mifos-data-import-tool-all-modules-populate-&-import

### PR DESCRIPTION
**Pull request by wkk91193**
This commit comprises of all the backend changes related to data import tool integration. The same structure has been conformed to, as of the  PR [office populate & Import](https://github.com/apache/fineract/pull/401)
which can be used for the purpose of referencing/sampling.Furthermore, the commit also includes integration tests for the main entitities Office,Loan,Savings,Client.
The frontend changes are can be viewed [here](https://github.com/openMF/community-app/pull/2504).

### CODE REVIEW

File: fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/bulkimport/importhandler/loan/LoanImportHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        Integer outcome_group_creation=groupHelper.createGroup(requestSpec,responseSpec,true);
+        Assert.assertNotNull("Could not create group" ,outcome_group_creation);
+
+        //in order to populate helper sheets
+        StaffHelper staffHelper=new StaffHelper();
+        Integer outcome_staff_creation =staffHelper.createStaff(requestSpec,responseSpec);
+        Assert.assertNotNull("Could not create staff",outcome_staff_creation);
+
+        LoanTransactionHelper loanTransactionHelper=new LoanTransactionHelper(requestSpec,responseSpec);
+        String jsonLoanProduct="{\"currencyCode\":\"USD\",\"includeInBorrowerCycle\":\"false\"," +
```
Comment (avikganguly01) [inline fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/bulkimport/importhandler/loan/LoanImportHandlerTest.java]: Use loan product test builder...avoid escaped string json

File: fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/bulkimport/importhandler/savings/SavingsImportHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        Integer outcome_group_creation=groupHelper.createGroup(requestSpec,responseSpec,true);
+        Assert.assertNotNull("Could not create group" ,outcome_group_creation);
+
+        //in order to populate helper sheets
+        StaffHelper staffHelper=new StaffHelper();
+        Integer outcome_staff_creation =staffHelper.createStaff(requestSpec,responseSpec);
+        Assert.assertNotNull("Could not create staff",outcome_staff_creation);
+
+        SavingsProductHelper savingsProductHelper=new SavingsProductHelper();
+        String jsonSavingsProduct="{\"currencyCode\":\"USD\",\"digitsAfterDecimal\":2," +
```
Comment (avikganguly01) [inline fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/bulkimport/importhandler/savings/SavingsImportHandlerTest.java]: Use SavingsProductHelper

File: fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/bulkimport/populator/loan/LoanWorkbookPopulatorTest.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        Integer outcome_group_creation=groupHelper.createGroup(requestSpec,responseSpec,true);
+        Assert.assertNotNull("Could not create group" ,outcome_group_creation);
+
+        //in order to populate helper sheets
+        StaffHelper staffHelper=new StaffHelper();
+        Integer outcome_staff_creation =staffHelper.createStaff(requestSpec,responseSpec);
+        Assert.assertNotNull("Could not create staff",outcome_staff_creation);
+
+        LoanTransactionHelper loanTransactionHelper=new LoanTransactionHelper(requestSpec,responseSpec);
+        String jsonLoanProduct="{\"currencyCode\":\"USD\",\"includeInBorrowerCycle\":\"false\"," +
```
Comment (avikganguly01) [inline fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/bulkimport/populator/loan/LoanWorkbookPopulatorTest.java]: Use LoanProductTestBuilder

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/center/CenterImportHandler.java [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+                if (meetings.get(i) != null)
+                    progressLevel = importCenterMeeting(result, i, dateFormat);
+                successCount++;
+                statusCell.setCellValue(TemplatePopulateImportConstants.STATUS_CELL_IMPORTED);
+                statusCell.setCellStyle(ImportHandlerUtils.getCellStyle(workbook, IndexedColors.LIGHT_GREEN));
+            } catch (AbstractPlatformDomainRuleException e) {
+                errorCount++;
+                e.printStackTrace();
+                errorMessage = e.getDefaultUserMessage();
+                writeCenterErrorMessage(centerId,errorMessage,progressLevel,statusCell,errorReportCell,row);
```
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/center/CenterImportHandler.java]: Move all the catch blocks to a functions in your Utils class and call it to get the error message. Call this function from all your import handler classes.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/chartofaccounts/ChartOfAccountsImportHandler.java [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+            accountTypeId=3L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }
+        else if(accountType!=null && accountType.equalsIgnoreCase(GLAccountType.INCOME.toString())){
+            accountTypeId=4L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }else if(accountType!=null && accountType.equalsIgnoreCase(GLAccountType.EXPENSE.toString())){
+            accountTypeId=5L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }
```
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/chartofaccounts/ChartOfAccountsImportHandler.java]: Move this code to GLAccountType new function fromString which will return the accountTypeEnum.

**Review by avikganguly01 — CHANGES_REQUESTED** [review]

**Comment by wkk91193** [comment]
@avikganguly01  Fixed those changes and ready for review!

**Pull request merged**